### ISSUE TITLE
Spellcheck excluded scopes don't work properly

### ISSUE THREAD

**Issue by Rahlir** [atom/spell-check#238]
*Edit by @rsese to add more specific repro steps*

1. Create a file called `test.md` with this content:

```md
# Hello Worrld
```

Notice at this point that `Worrld` is shown as misspelled.

2. In `Settings > Packages > spell-check > Excluded Scopes`, add `markup.heading.heading-1.gfm`

At this point, `Worrld` is no longer shown as misspelled.

3. Restart Atom.

Expect `Worrld` to not be shown as misspelled, actual behavior is it is shown as misspelled.

If you go back to `Settings > Packages > spell-check > Excluded Scopes`, and remove `markup.heading.heading-1.gfm` and then paste it right back in, `Worrld` is not shown as misspelled anymore.

Tested with 1.27.0-dev-14e6500e0 on macOS 10.12.6.

---

#### Description

When I setup exceptions for Latex scopes in the new Atom Beta release and it works fine, until I close Atom and open it again. The spell check suddenly goes back to checking everything and in order for the exceptions to work, I have to delete and paste back the scopes I don't want checked.

Plus, and this might be a separate issue, adding "known words" in the spell-check also doesn't work, the known words are still labeled as incorrect.

My excluded scopes: support.function.general.tex, support.function.textbf.latex, storage.type.function.latex, support.function.textit.latex, meta.preamble.latex, meta.embedded.block.source

### Steps to Reproduce

1. Set up excluded scopes
2. Quit Atom
3. Re-open Atom and a relevant file

**Expected behavior:** Text within excluded scopes should be ignored

**Actual behavior:** Text is spell-checked even though it is within excluded scope

**Reproduces how often:** always so far

### Versions

1.25.0-beta2

### Additional Information

Any additional information, configuration or data that might be necessary to reproduce the issue.

**Comment by rsese** [comment]
Thanks for the report - can you share a minimal specific example with an example file, a single excluded scope, and where the text is spell-checked after restarting Atom?

**Comment by gncs** [comment]
I am experiencing the exact same issue as @Rahlir. My excluded scopes are `      "keyword.control.preamble.latex"
      "keyword.control.cite.latex"
      "support.function.general.tex"
      "support.function.textbf.latex"
      "support.function.texttt.latex"
      "support.function.textit.latex"
      "constant.other.general.math.tex"
      "constant.character.math.tex"
`.

**Comment by gncs** [comment]
Example (together with the configuration mentioned above):
```latex
\documentclass[a4paper,10pt]{scrartcl}

\usepackage[utf8]{inputenc}
\usepackage[T1]{fontenc}
\usepackage{lmodern}

\usepackage[backend=biber]{biblatex}
\addbibresource{references.bib}

\begin{document}
\end{document}
```

**Comment by rsese** [comment]
Thanks a lot @gncs :+1:, I see what folks are saying now - I can reproduce the situation you described @gncs and can also repro with this Markdown example in safe mode:

1. Create a file called `test.md` with this content:

```md
# Hello Worrld
```

Notice at this point that `Worrld` is shown as misspelled.

2. In `Settings > Packages > spell-check > Excluded Scopes`, add `markup.heading.heading-1.gfm`

At this point, `Worrld` is no longer shown as misspelled.

3. Restart Atom.

Expect `Worrld` to not be shown as misspelled, actual behavior is it is shown as misspelled.

If you go back to `Settings > Packages > spell-check > Excluded Scopes`, and remove `markup.heading.heading-1.gfm` and then paste it right back in, `Worrld` is not shown as misspelled anymore.

**Comment by lee-dohm** [comment]
@maxbrunsfeld Can you take a look at this?

**Comment by werunom** [comment]
Experiencing similar problem in latex with the following excluded scopes - `.support.function.latex` and  `.keyword.control.latex`

**Comment by maxbrunsfeld** [comment]
Hey everyone, sorry for putting out a totally buggy implementation of this long-requested feature. I believe the bugs have been addressed. The fix will go out this week in Atom 1.26.1.
